flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/NettyShuffleUtils.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.shuffle;
+
+import org.apache.flink.runtime.deployment.ResultPartitionDeploymentDescriptor;
+import org.apache.flink.util.Preconditions;
+
+/**
+ * Common utility methods for default Netty implementation of Shuffle Service.
+ */
+public enum NettyShuffleUtils {
+	;
+
+	public static void checkReleaseOnConsumptionIsSupportedForPartition(PartitionDescriptor partitionDescriptor) {
+		checkReleaseOnConsumptionIsSupportedForPartition(
+			partitionDescriptor.isReleasedOnConsumption(),
+			partitionDescriptor.getPartitionType().isBlocking());
+	}
+
+	public static void checkReleaseOnConsumptionIsSupportedForPartition(
+			ResultPartitionDeploymentDescriptor resultPartitionDeploymentDescriptor) {
+		checkReleaseOnConsumptionIsSupportedForPartition(
+			resultPartitionDeploymentDescriptor.isReleasedOnConsumption(),
+			resultPartitionDeploymentDescriptor.getPartitionType().isBlocking());
+	}
+
+	public static void checkReleaseOnConsumptionIsSupportedForPartition(
+			boolean isReleasedOnConsumption,
+			boolean isBlockingPartition) {
+		Preconditions.checkArgument(
+			!isReleasedOnConsumption || isBlockingPartition,
+			"Auto-release on consumption is supported only for blocking partition types");
```

Comment:
I'd invert this to say "Auto-release on consumption must not be disabled for pipelined partition types." The current message is a bit confusing; after all auto-release on consumption is _effectively_ supported for pipelined partitions.